### DISCUSSION TITLE
What's the best way to persist a session across app opens?

### DISCUSSION THREAD

**Discussion by akshatksharma** [MasterJ93/ATProtoKit#71] [Q&A]
Hey all! Was wondering what was the preferred way to make it so that the user doesn't need to reauthenticate each time the app opens. Up until v0.21 I was persisting the UserSession on disk and passing that into the `ATProtoKit` initializer. However now that we only talk to the UserSession via the SessionConfiguration, this approach no longer works.

Is there something else I should be doing? A bit of a noob so apologies if this question is not clear

**Comment by MasterJ93** [comment]
Sorry: I don't know why I don't get notifications from here, so I missed this question.

At this time, you can use `ATProtocolConfiguration.session` and get the access or refresh tokens that way. No need to constantly use `authenticate()`. However, it does seem like this is causing some issues, so I'm working on trying to make a better way of doing this that doesn't disrupt the ways developers want to write their code. I'm making an interim solution that I'll hopefully make this week, and then I'll think of a more stable solution later on.

Sorry for the mess.
